src/apps/external_executor/protobuf/historical.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+import "kv.proto";
+
+package externalexecutor.protobuf;
+
+option optimize_for = LITE_RUNTIME;
+
+service Historical
+{
+  // Historical data look up
+  rpc GetHistoricalData(HistoricalData) returns (QueryResponse) {}
+}
+
+message HistoricalData
+{
+  string map_name = 1;
+  bytes key = 2;
+  int32 view = 3;
+  int32 seqno = 4;
+}
+
+message RetryQuery { bool retryquery = 1; }
+message QueryResponse
+{
+  externalexecutor.protobuf.KVValue data = 1;
+  RetryQuery retry = 2;
+}
```

Comment:
Note: Naming of this service will be eventually merged with the indexing service